### ISSUE TITLE
Deleted attributes destroy export

### ISSUE THREAD

**Issue by macwinnie** [Attendize/Attendize#215]
Deleting orders / attendees over the flags within the database causes the export of CSV-Files over the survey within the backend to fail.

**Comment by daveearley** [comment]
Hi @macwinnie ,

Could I get more information on this? Are you saying that when you manually delete an order or attendee from the DB the exporting of survey questions fails?

**Comment by macwinnie** [comment]
Hi @daveearley =)

I searched for a proper way for deleting attendees from an event – especially test-attendees. Within the UI, there I could not find a possibility for really deleting them from showing up within the attendees-list, so I used the db-attribute `deleted_at`  within the `orders`-table and determined that this does the job.
Some time later, a colleague wanted to export the attendees over the UI, but it crashed. With looking at the error.log-File, I determined that something crashed with the deleted_at-Attribute ... Reproduced the error for you, you can find the log-entry below.

Cheers,
Martin

```
[2016-10-22 20:01:59] production.ERROR: ErrorException: Trying to get property of non-object in /var/www/attendize/app/Models/Event.php:255
Stack trace:
#0 /var/www/attendize/app/Models/Event.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(8, 'Trying to get p...', '/var/www/attend...', 255, Array)
#1 /var/www/attendize/bootstrap/cache/compiled.php(11335): App\\Models\\Event->getSurveyAnswersAttribute(NULL)
#2 /var/www/attendize/bootstrap/cache/compiled.php(11295): Illuminate\\Database\\Eloquent\\Model->mutateAttribute('survey_answers', NULL)
#3 /var/www/attendize/bootstrap/cache/compiled.php(11287): Illuminate\\Database\\Eloquent\\Model->getAttributeValue('survey_answers')
#4 /var/www/attendize/bootstrap/cache/compiled.php(11627): Illuminate\\Database\\Eloquent\\Model->getAttribute('survey_answers')
#5 /var/www/attendize/app/Http/Controllers/EventSurveyController.php(272): Illuminate\\Database\\Eloquent\\Model->__get('survey_answers')
#6 [internal function]: App\\Http\\Controllers\\EventSurveyController->App\\Http\\Controllers\\{closure}(Object(Maatwebsite\\Excel\\Classes\\LaravelExcelWorksheet))
#7 /var/www/attendize/vendor/maatwebsite/excel/src/Maatwebsite/Excel/Writers/LaravelExcelWriter.php(223): call_user_func(Object(Closure), Object(Maatwebsite\\Excel\\Classes\\LaravelExcelWorksheet))
#8 /var/www/attendize/app/Http/Controllers/EventSurveyController.php(278): Maatwebsite\\Excel\\Writers\\LaravelExcelWriter->sheet('survey_answers_...', Object(Closure))
#9 [internal function]: App\\Http\\Controllers\\EventSurveyController->App\\Http\\Controllers\\{closure}(Object(Maatwebsite\\Excel\\Writers\\LaravelExcelWriter))
#10 /var/www/attendize/vendor/maatwebsite/excel/src/Maatwebsite/Excel/Excel.php(87): call_user_func(Object(Closure), Object(Maatwebsite\\Excel\\Writers\\LaravelExcelWriter))
#11 /var/www/attendize/bootstrap/cache/compiled.php(6327): Maatwebsite\\Excel\\Excel->create('answers-as-of-2...', Object(Closure))
#12 /var/www/attendize/app/Http/Controllers/EventSurveyController.php(279): Illuminate\\Support\\Facades\\Facade::__callStatic('create', Array)
#13 [internal function]: App\\Http\\Controllers\\EventSurveyController->showExportAnswers(Object(Illuminate\\Http\\Request), '3', 'xls')
#14 /var/www/attendize/bootstrap/cache/compiled.php(9424): call_user_func_array(Array, Array)
#15 /var/www/attendize/bootstrap/cache/compiled.php(9486): Illuminate\\Routing\\Controller->callAction('showExportAnswe...', Array)
#16 /var/www/attendize/bootstrap/cache/compiled.php(9466): Illuminate\\Routing\\ControllerDispatcher->call(Object(App\\Http\\Controllers\\EventSurveyController), Object(Illuminate\\Routing\\Route), 'showExportAnswe...')
#17 [internal function]: Illuminate\\Routing\\ControllerDispatcher->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#18 /var/www/attendize/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#19 [internal function]: Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#20 /var/www/attendize/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#21 /var/www/attendize/bootstrap/cache/compiled.php(9467): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#22 /var/www/attendize/bootstrap/cache/compiled.php(9454): Illuminate\\Routing\\ControllerDispatcher->callWithinStack(Object(App\\Http\\Controllers\\EventSurveyController), Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request), 'showExportAnswe...')
#23 /var/www/attendize/bootstrap/cache/compiled.php(8524): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request), 'App\\\\Http\\\\Contro...', 'showExportAnswe...')
#24 /var/www/attendize/bootstrap/cache/compiled.php(8511): Illuminate\\Routing\\Route->runController(Object(Illuminate\\Http\\Request))
#25 /var/www/attendize/bootstrap/cache/compiled.php(8225): Illuminate\\Routing\\Route->run(Object(Illuminate\\Http\\Request))
#26 [internal function]: Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#27 /var/www/attendize/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#28 /var/www/attendize/app/Http/Middleware/FirstRunMiddleware.php(37): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#29 [internal function]: app\\Http\\Middleware\\FirstRunMiddleware->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#30 /var/www/attendize/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#31 [internal function]: Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#32 /var/www/attendize/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#33 /var/www/attendize/app/Http/Middleware/Authenticate.php(27): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#34 [internal function]: App\\Http\\Middleware\\Authenticate->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#35 /var/www/attendize/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#36 [internal function]: Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#37 /var/www/attendize/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#38 [internal function]: Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#39 /var/www/attendize/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#40 /var/www/attendize/bootstrap/cache/compiled.php(8226): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#41 /var/www/attendize/bootstrap/cache/compiled.php(8217): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#42 /var/www/attendize/bootstrap/cache/compiled.php(8207): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#43 /var/www/attendize/bootstrap/cache/compiled.php(2419): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#44 [internal function]: Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#45 /var/www/attendize/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\\Http\\Request))
#46 /var/www/attendize/app/Http/Middleware/GeneralChecks.php(25): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#47 [internal function]: app\\Http\\Middleware\\GeneralChecks->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#48 /var/www/attendize/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
```

**Comment by daveearley** [comment]
Merged with #133
